src/backend/distributed/planner/multi_explain.c

```diff
@@ -293,10 +296,60 @@ RemoteExplain(Task *task, ExplainState *es)
 		ShardPlacement *taskPlacement = list_nth(taskPlacementList, placementIndex);
 		char *nodeName = taskPlacement->nodeName;
 		uint32 nodePort = taskPlacement->nodePort;
+		MultiConnection *connection = NULL;
+		PGresult *queryResult = NULL;
+		int connectionFlags = SESSION_LIFESPAN;
+		int executeResult = 0;
 
 		remotePlan->placementIndex = placementIndex;
-		remotePlan->explainOutputList = ExecuteRemoteQuery(nodeName, nodePort,
-														   NULL, explainQuery);
+
+		connection = GetNodeConnection(connectionFlags, nodeName, nodePort);
```

Comment:
We should probably use GetPlacementConnection here. We might create a deadlock if we run an EXPLAIN on the wrong connection after a DDL.